semgrep/semgrep/test.py

```diff
@@ -79,50 +94,32 @@ def _test_compute_confusion_matrix() -> None:
     assert fn == 2
 
 
+def _annotations(annotation: str) -> Set[str]:
+    # returns something like: {"#ruleid:", "# ruleid:", "//ruleid:", ...}
+    return {
+        f"{comment_syntax[0]}{space}{annotation}"
+        for comment_syntax, space in product(COMMENT_SYNTAXES, SPACE_OR_NO_SPACE)
+    }
+
+
 def line_has_todo_rule(line: str) -> bool:
-    return (
-        "#todoruleid:" in line
-        or "# todoruleid:" in line
-        or "// todoruleid:" in line
-        or "//todoruleid:" in line
-        or "(*todoruleid:" in line
-        or "(* todoruleid:" in line
-    )
+    todo_rule_annotations = _annotations(TODORULEID)
+    return any([annotation in line for annotation in todo_rule_annotations])
```

Comment:
Noice, I love all the code removal.

Fun Pythonism/nitpick: using a generator instead of a list in `any`/`all` calls will be more memory efficient and produce the same result. E.g. `any(i for i in iterable)` vs. `any([i for i in iterable])`. Since these lists are small it won't have much of an effect, but for larger lists/datasets it will. This is because the list comprehension builds the whole list in-memory, then calls `any`. A generator only holds a single element in memory at a time, and since `any`/`all` will exit early if they can it becomes quite a bit more efficient :+1: 

Not a huge deal here, but using generator statements with these calls will always be more efficient with no downside.

Reply:
Sounds like a good CR-style Semgrep rule ;)